### ISSUE TITLE
Manual does not document slices nor `..^`

### ISSUE THREAD

**Issue by snej** [nim-lang/Nim#14671]
The Nim manual does not describe the slice type nor slicing arrays/sequences; the only documentation of it I've seen is [in in the tutorial](https://nim-lang.org/docs/tut1.html#advanced-types-slices).

The manual also never explains what `..^` means, even though it uses it in several examples.

**Comment by kaushalmodi** [comment]
See https://nim-lang.github.io/Nim/system.html#%5E.t%2Cint

Note that the correct way is putting a space between `..` and `^`. For `[a .. ^b]`, `^b` is a separate entity and `..` is the binary slice operator https://nim-lang.github.io/Nim/system.html#..%2C%2C .

**Comment by solo989** [comment]
Both .. ^ and ..^ are correct.
..^ is a shortcut for .. ^.

**Comment by Araq** [comment]
The manual doesn't claim to describe the standard library.

**Comment by kaushalmodi** [comment]
I realized that I forgot to submit this as a PR yesterday. @Araq I see that you already closed this issue, but still I submitted that PR as I had it ready anyways.

**Comment by snej** [comment]
> The manual doesn't claim to describe the standard library.

That makes sense ... but to the average developer there's a blurry line between a language and its library when the language has this much syntactic flexibility. In this case the `..` syntax is part of the language when describing ordinal types or case ranges, but not when accessing collections; I wasn't aware of that.

And the manual does inevitably touch on things that are defined in the library, like `echo`, `$()`, `@[...]` and so on.

Perhaps there should be a manual for the core modules of the standard library. The auto-generated docs for `system` are hard to follow because they're broken into hundreds of functions and operators ... I would never have discovered the `^` operator if I didn't know it existed.

**Comment by Araq** [comment]
That's true but significant effort went into improving system.nim's documentation. See https://nim-lang.org/docs/system.html#system-module-misc maybe it should list the ``^`` notation too. Ping @narimiran

**Comment by timotheecour** [comment]
2 things would improve discoverability:
* make doc search understand `^` without backticks:
searching for this in dochack search returns nothing:
```
^
```

it works with backicks
```
`^`
```

* add an operator section (which can overlap with procs/templates etc; it not mutually disjoint sections)

enhancing https://nim-lang.org/docs/system.html#system-module-misc is fine, but doing those 2 things would benefit automatically all modules (including user modules)
